packages/events/src/once/sync.rs

```diff
@@ -0,0 +1,518 @@
+where
+    T: Send,
+{
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// A sender that can send a value through a thread-safe event.
+///
+/// The sender owns the underlying channel and can be moved across threads.
+/// After calling [`send`](EventSender::send), the sender is consumed.
+#[derive(Debug)]
+pub struct EventSender<T>
+where
+    T: Send,
+{
+    sender: Option<oneshot::Sender<T>>,
+}
+
+impl<T> EventSender<T>
+where
+    T: Send,
+{
+    /// Sends a value through the event.
+    ///
+    /// This method consumes the sender and always succeeds, regardless of whether
+    /// there is a receiver waiting.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use events::once::Event;
+    ///
+    /// let event = Event::<i32>::new();
+    /// let sender = event.sender();
+    /// sender.send(42);
+    /// ```
+    pub fn send(mut self, value: T) {
+        if let Some(sender) = self.sender.take() {
+            // We don't care if the receiver is dropped - sending always succeeds
+            drop(sender.send(value));
+        }
+    }
+}
+
+// SAFETY: EventSender can be sent across threads when T: Send, which is already required
+unsafe impl<T: Send> Send for EventSender<T> {}
+// SAFETY: EventSender can be shared across threads when T: Send
+unsafe impl<T: Send> Sync for EventSender<T> {}
+
+/// A receiver that can receive a value from a thread-safe event.
+///
+/// The receiver owns the underlying channel and can be moved across threads.
+/// After calling [`receive`](EventReceiver::receive), the receiver is consumed.
+#[derive(Debug)]
+pub struct EventReceiver<T>
+where
+    T: Send,
+{
+    receiver: Option<oneshot::Receiver<T>>,
+}
+
+impl<T> EventReceiver<T>
+where
+    T: Send,
+{
+    /// Receives a value from the event.
+    ///
+    /// This method consumes the receiver and waits for a sender to send a value.
+    /// If the sender is dropped without sending, this method will wait forever.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use events::once::Event;
+    ///
+    /// let event = Event::<i32>::new();
+    /// let (sender, receiver) = event.endpoints();
+    ///
+    /// sender.send(42);
+    /// let value = receiver.receive();
+    /// assert_eq!(value, 42);
+    /// ```
+    #[must_use]
+    pub fn receive(mut self) -> T {
+        self.receiver.take().map_or_else(
+            || unreachable!("receiver should always be Some when receive is called"),
+            |receiver| {
+                receiver.recv().unwrap_or_else(|_| {
+                    // If the sender was dropped without sending, we wait forever
+                    // as per the specification
+                    loop {
+                        std::thread::park();
+                    }
+                })
+            },
+        )
+    }
+}
+
+// SAFETY: EventReceiver can be sent across threads when T: Send, which is already required
+unsafe impl<T: Send> Send for EventReceiver<T> {}
```

Comment:
Are Send/Sync not automatically derived here by the compiler? If there is a reason we need to explicitly declare them, let's document the justification with inline comments.

The safety comments here do not appear to conform to the repository coding guidelines.